wire-ios-sync-engine/Source/UserSession/ZMUserSession/AppVersionMigrations/AppVersionMigration_4_2_0.swift

```diff
@@ -20,20 +20,43 @@ import Foundation
 import WireDomain
 
 /// **Issue:**: To simplify the logic, we rely solely on journal value to perform InitialSync or not
-final class AppVersionMigration_4_2_0: AppVersionMigration {
+struct AppVersionMigration_4_2_0: AppVersionMigration {
 
+    let appGroupIdentifier: String?
     let lastEventIDRepository: LastEventIDRepositoryInterface
     var journal: JournalProtocol
     let version: SemanticVersion = "4.2.0"
 
-    init(lastEventIDRepository: LastEventIDRepositoryInterface, journal: JournalProtocol) {
-        self.lastEventIDRepository = lastEventIDRepository
-        self.journal = journal
-    }
-
     func perform() async throws {
+
         if lastEventIDRepository.fetchLastEventID() == nil {
             journal[.isInitialSyncRequired] = true
         }
+
+        deleteOldLogFiles()
```

Comment:
I'm raising this one more time: This code won't run for anybody who has already updated to a 4.2.0 Beta.